osu.Game/Graphics/UserInterface/OsuModSettingsNumberBox.cs

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class OsuModSettingsNumberBox : OsuModSettingsTextBox
+    {
+        protected override bool CanAddCharacter(char character) => char.IsNumber(character);
+    }
+}
```

Comment:
This can be move into `ModRandom` as protected class.

Currently only `ModRandom` and its `{Ruleset}ModRandom` use this setting box.

Reply:
I don't agree with this, this is not a textbox specifically for `ModRandom`, it should be in the `Mods` namespace if it's scope is still for mods (which I believe should not be, based on https://github.com/ppy/osu/pull/13463#discussion_r650341664).

Reply:
Yep, move it to `Mods` namespace sounds better than in current namespace.

Reply:
Since both classes are now nested in `SettingsTextBox` and `SettingsNumberBox` this doesn't matter anymore.